### PR TITLE
Remove MapSector completely.

### PR DESCRIPTION
**Pull request by lhofhansl**
This "just" removes MapSector, which didn't fulfill any purpose anymore.
No change of functionality. It should make it is easier to improve multi-threading in the future.

Performance is unchanged as far as I can tell.
While in there I also remove a few #IF 0 sections.

### CODE REVIEW

File: src/client/clientmap.cpp [hunk 1]
```diff
@@ -153,77 +136,54 @@ void ClientMap::updateDrawList()
...
-
-		MapBlockVect sectorblocks;
-		sector->getBlocks(sectorblocks);
-
+	for (auto &it : m_blocks) {
+		MapBlock *block = it.second;
 		/*
-			Loop through blocks in sector
+		  Compare block position to camera position, skip
+		  if not seen on display
```
Comment (sfan5) [inline src/client/clientmap.cpp:143]: Mixed indentation

File: src/map.cpp [hunk 2]
```diff
@@ -338,59 +326,46 @@ void Map::timerUpdate(float dtime, float unload_timeout, u32 max_loaded_blocks,
...
+			MapBlock *block = it.second;
+
+			block->incrementUsageTimer(dtime);
+
+			if (block->refGet() == 0
+			    && block->getUsageTimer() > unload_timeout) {
+				v3s16 p = block->getPos();
+				// Save if modified
+				if (block->getModified() != MOD_STATE_CLEAN
+				    && save_before_unloading) {
```
Comment (sfan5) [inline src/map.cpp:340]: Pretty sure this & the condition above now fit on a single line

**Comment by sfan5** [comment]
At one point I wondered what purpose MapSectors fulfilled but couldn't think of something. Thinking about it more, I'm pretty sure they are leftovers from when the map was stored as a structure of directories and small files (cd35949cd07dff9f64f982b7a28999b89dfbeb5a).
So it does look like these can be safely removed.

**Comment by lhofhansl** [comment]
The Android test failure looks legit, the VS ones don't.

**Comment by lhofhansl** [comment]
Fixed Android makefile and remove some remnants of MapSector (whitelist file)

**Comment by lhofhansl** [comment]
If when this is merged I will look into multi-threading the server (and possibly the client) better.

**Comment by lhofhansl** [comment]
Hmm... I do notice that now some more CPU is spent in the `Map<v3s16, MapBlock*>` in cases with large viewing_ranges.
So the extra indirection that `MapSector` provided helped some. (RTree now shows up prominently in the profiler.)

I tried using `unordered_map` along with a good hash function, but that made it even worse.

Shame I actually spend a few hours on this, and it looks like it's hurting performance.

**Comment by SmallJoker** [comment]
How much more CPU? Did you measure both execution times in the profiler? How much is the difference?
Seeing how much useless code there is, I believe it's possible to accept that as a trade-off.

**Comment by lhofhansl** [comment]
Closing this one. Good try :)

**Pull request closed without merging**